### ISSUE TITLE
How to find out which Mailbox is missing with error 'No such mailbox'

### ISSUE THREAD

**Issue by StefanSchroeder** [neomutt/neomutt#4424]
I am using neomutt for a few days. It generally works fine. But every ~5 seconds an error message appears:

`IMAP command fails: aXXX No [TRYCREATE] No such mailbox`

where XXX is an increasing number.

I have not configured in my neomuttrc any non-existent mailbox. Why this message? How do I find out which mailbox is missing? How to investigate?

I tried `neomutt -d2` and reviewed `.neomuttdebug0`. The error message appears there but gives no indication what the missing mailbox is.

Any help appreciated. I feel that the error might be the reason for a slow UI.

**Comment by roccoblues** [comment]
> `IMAP command fails: aXXX No [TRYCREATE] No such mailbox`

Could this be `IMAP command failed: ...`? I can't find the other string in our code.

> How to investigate?
> I tried `neomutt -d2` and reviewed `.neomuttdebug0`. The error message appears there but gives no indication what the missing mailbox is.

I don't use IMAP so I can't help much. Have you tried increasing the debug level? You can go up to `-d5`. But please don't share without reviewing as higher level can contain private information.

**Comment by StefanSchroeder** [comment]
HELO roccoblues, thank you for taking the time to look into it.

You are right. It's `IMAP command failed:` I mistyped.

This is some context around the error `a0520`. I cannot see anything suspicious.

```
[2024-11-11 23:49:06]<5> mutt_socket_readln_d() 5< a0518 OK STATUS completed
[2024-11-11 23:49:06]<5> mutt_socket_readln_d() 5< * STATUS Private (MESSAGES 34 RECENT 0 UIDNEXT 35 UIDVALIDITY 1122461104 UNSEEN 0)
[2024-11-11 23:49:06]<3> cmd_parse_status() Private (UIDVALIDITY: 1122461104, UIDNEXT: 35) 34 messages, 0 recent, 0 unseen
[2024-11-11 23:49:06]<3> cmd_parse_status() Running default STATUS handler
[2024-11-11 23:49:06]<3> cmd_parse_status() Found Private in mailbox list (OV: 1122461104 ON: 35 U: 0)
[2024-11-11 23:49:06]<5> sb_mailbox_observer() mailbox done, request WA_RECALC
[2024-11-11 23:49:06]<5> mutt_socket_readln_d() 5< a0519 OK STATUS completed
[2024-11-11 23:49:06]<5> mutt_socket_readln_d() 5< a0520 NO [TRYCREATE] No such mailbox.
[2024-11-11 23:49:06]<M> imap_cmd_step() IMAP command failed: a0520 NO [TRYCREATE] No such mailbox.
[2024-11-11 23:49:06]<5> msgwin_set_text() MW SET: 58, IMAP command failed: a0520 NO [TRYCREATE] No such mailbox.
[2024-11-11 23:49:06]<5> msgwin_calc_rows() msgwin_calc_rows() => 1
[2024-11-11 23:49:06]<5> msgwin_recalc() recalc done, request WA_REPAINT
[2024-11-11 23:49:06]<5> msgwin_repaint() msgwin repaint done
[2024-11-11 23:49:09]<5> mutt_socket_readln_d() 5< * STATUS Mail.Trash (MESSAGES 0 RECENT 0 UIDNEXT 63 UIDVALIDITY 1126375861 UNSEEN 0)
[2024-11-11 23:49:09]<3> cmd_parse_status() Mail.Trash (UIDVALIDITY: 1126375861, UIDNEXT: 63) 0 messages, 0 recent, 0 unseen
[2024-11-11 23:49:09]<3> cmd_parse_status() Running default STATUS handler
[2024-11-11 23:49:09]<3> cmd_parse_status() Found Mail.Trash in mailbox list (OV: 1126375861 ON: 63 U: 0)
[2024-11-11 23:49:09]<5> sb_mailbox_observer() mailbox done, request WA_RECALC
[2024-11-11 23:49:09]<5> mutt_socket_readln_d() 5< a0521 OK STATUS completed
```

**Comment by gahr** [comment]
Run with -d10. You should see the command we send (same Axxx but opposite direction).

My wild bet would be on the "Mail" mailbox. Could it be that you have a number of Mail.xyz folders but not Mail by itself? This is possible, IMAP-wise.

**Comment by roccoblues** [comment]
> Run with -d10. You should see the command we send (same Axxx but opposite direction).
TIL 😄 

Comparing this:
```
[2024-11-11 23:49:06]<3> cmd_parse_status() Private (UIDVALIDITY: 1122461104, UIDNEXT: 35) 34 messages, 0 recent, 0 unseen
[2024-11-11 23:49:06]<3> cmd_parse_status() Running default STATUS handler
[2024-11-11 23:49:06]<3> cmd_parse_status() Found Private in mailbox list (OV: 1122461104 ON: 35 U: 0)
[2024-11-11 23:49:06]<5> sb_mailbox_observer() mailbox done, request WA_RECALC
[2024-11-11 23:49:06]<5> mutt_socket_readln_d() 5< a0519 OK STATUS completed
[2024-11-11 23:49:06]<5> mutt_socket_readln_d() 5< a0520 NO [TRYCREATE] No such mailbox.
```

and this:
```
[2024-11-11 23:49:09]<3> cmd_parse_status() Mail.Trash (UIDVALIDITY: 1126375861, UIDNEXT: 63) 0 messages, 0 recent, 0 unseen
[2024-11-11 23:49:09]<3> cmd_parse_status() Running default STATUS handler
[2024-11-11 23:49:09]<3> cmd_parse_status() Found Mail.Trash in mailbox list (OV: 1126375861 ON: 63 U: 0)
```

Do you have a mailbox called `Private`?

**Comment by gahr** [comment]
> STATUS Private (MESSAGES 34 RECENT 0 UIDNEXT 35 UIDVALIDITY 1122461104 UNSEEN 0)

Yeah, `Private` definitely exists

**Comment by gahr** [comment]
> > Run with -d10. You should see the command we send (same Axxx but opposite direction).
> > My wild bet would be on the "Mail" mailbox. Could it be that you have a number of Mail.xyz folders but not Mail by itself? This is possible, IMAP-wise.
> 
> I will check this. OTOH if it's possible IMAP-wise it shouldn't raise an error, should it?

I mean it's possible to have Foo.Bar but not Foo. The error is you being subscribed to Foo.

**Comment by gahr** [comment]
> Run with -d10. You should see the command we send (same Axxx but opposite direction).

Sorry, I was confused. You should already see messages *to* your IMAP server with `-d5`. Look for `5>`.

**Comment by StefanSchroeder** [comment]
Ok, so what's my next step here if gahr's assessment is correct.

```
set spoolfile = +INBOX
set record = +Sent%20Items
set trash = +Trash
set postponed = +Drafts
mailboxes =INBOX =Sent%20Items =Trash =Drafts
```

I guess this is the relevant section in my neomuttrc. 

I am not subscribing to 'Private', yet it shows up in the folders. Obviously the existing mailboxes are being queried and fetched from the IMAP server.

Interestingly, there is mention of 'Trash' as well as 'Mail.Trash' in the logs.

How do I determine is correct nomenclature to refer to my 'Trash'?

**Comment by gahr** [comment]
So you have `imap_check_subscribed` set? If so, NeoMutt will query the list of subscribed mailboxes.

Also, I can't check the code right now but I'm not sure about the url-escape in Send%2Items. Can you try with "+Sent Items"?

Anyway, please look at the STATUS commands we send. See if you spot a mailbox that shouldn't be there.

**Comment by StefanSchroeder** [comment]
I have indeed set `imap_check_subscribed`.

I don't think the percent-encoding is a problem, I found this hint in this ticket: https://github.com/neomutt/neomutt/issues/2225

But I will try anyway. Thanks for the tip.

**Comment by StefanSchroeder** [comment]
Re. 1. Right, and we try to figure out what it is and why I am subscribed to it when it doesn't exist.
Re. 2. True
Re. 3. Ok, NeoMutt asks for a list of subscribed mailboxes, fair enough.
Re. 4. Why would the server tell me that a mailbox doesn't exist when it previously told me that I am subscribed to it
Re. 5. this is a consequence of 1-4
Re. 6. The IMAP server says it doesn't know about it when it previously told me that I am subscribed to it (see 4)
Re. 7. fine. But why not telling me which mailbox it cannot find.

Looking at the logs with d6 (which is the highest level), there is a suspicious mailbox

```
2024-11-14 20:00:31]<3> cmd_parse_lsub() Subscribing to Mail.Trash
[2024-11-14 20:00:31]<1> parse_rc_buffer() NT_COMMAND: mailboxes
[2024-11-14 20:00:31]<1> mailbox_add() Adding mailbox: 'imaps://XXXX@example.com@imap.example.com:993/Mail.Trash' label '[NONE]', poll [UNSPECIFIED], notify [UNSPECIFIED]
```

The fishy one is Mail.Trash because my Trash folder is named Trash and not Mail.Trash. But what makes the IMAP server telling me that I am subscribed to a folder Mail.Trash when it doesn't exist?

**Comment by StefanSchroeder** [comment]
Now I found out another interesting thing. When using Thunderbird to work with that IMAP server, the Mail.Trash folder did not show up. But then I changed the configuration to show all folder, not only subscribed ones. Here now Mail.Trash DID show up in the folder list. But when I tried to delete it, the command failed because the folder didn't exist. 
I come to conclude it must be some sort of virtual default folder. I will look at the IMAP configuration at the hoster to check.
